tests/generate-testdata.js

```diff
@@ -0,0 +1,335 @@
+// signature suite implementation to use for issuing verifiable credentials.
+const suitePromise = (async () => {
+  const { Ed25519VerificationKey2018 } = await import('@digitalbazaar/ed25519-verification-key-2018');
+  const { Ed25519Signature2018 } = await import('@digitalbazaar/ed25519-signature-2018');
+
+  const seed = Buffer.from('9b937b81322d816cfab9d5a3baacc9b2');
+  const keyPair = await Ed25519VerificationKey2018.generate({ seed });
+
+  keyPair.id = `did:key:${keyPair.fingerprint()}#${keyPair.fingerprint()}`;
+  keyPair.controller = `did:key${keyPair.fingerprint()}`;
+  return new Ed25519Signature2018({ key: keyPair });
+})();
+
+// noopMutator returns the input without any mutation.
+const noopMutator = (input) => klona(input);
+
+// mutatingDocumentLoader is a document loader implementation that allows mutation of
+// the https://www.w3.org/2018/credentials/v1 context via a given mutation
+// function.
+const mutatingDocumentLoader = async ({ mutate = noopMutator } = {}) => {
+  const document = mutate(CONTEXT);
+  return async (documentUrl) => {
+    if (documentUrl === CONTEXT_URL) {
+      return { contextUrl: null, documentUrl, document };
+    }
+    return strictDocumentLoader(documentUrl);
+  };
+};
+
+// w3cDate converts a given date-like argument into W3C datetime format with
+// seconds resolution.
+const w3cDate = (date) => {
+  let d = date;
+  if (date === undefined || date === null) {
+    d = new Date();
+  } else if (typeof date === 'number' || typeof date === 'string') {
+    d = new Date(date);
+  }
+  const str = d.toISOString();
+  return `${str.substr(0, str.length - 5)}Z`;
+};
+
+const addProof = async ({ credentialMutator = noopMutator, contextMutator = noopMutator } = {}) => {
+  const suite = await suitePromise;
+  const document = credentialMutator(require('./valid-credential.json'));
+  const documentLoader = await mutatingDocumentLoader({ mutate: contextMutator });
+
+  let proof = {
+    // Using a specific date allows proof creation to be idempotent.
+    created: w3cDate('2006-01-02T15:04:05Z'),
+    verificationMethod: suite.verificationMethod,
+    proofPurpose: 'assertionMethod',
+    type: 'Ed25519Signature2018',
+  };
+
+  const verifyData = await suite.createVerifyData({
+    document,
+    proof,
+    documentLoader,
+  });
+
+  proof = await suite.sign({ verifyData, proof });
+  jsonld.addValue(document, 'proof', proof);
+  return document;
+};
+
+(async () => {
+  const sampleVCs = new Map();
+
+  {
+    const description = 'valid_vc';
+    sampleVCs.set(description, await addProof());
+  }
+
+  {
+    const description = 'vc:@context:missing';
+    const credentialMutator = (credential) => {
+      const doc = klona(credential);
+      delete doc['@context'];
+      return doc;
+    };
+    sampleVCs.set(description, await addProof({ credentialMutator }));
+  }
+
+  {
+    const invalidValues = new Map([
+      // ['vc:@context:boolean', false],
```

Comment:
These are commented out because of the difficulty of testing with changes to context right?

Reply:
Correct, any commented out test cases were ones I was not able to generate valid signatures for. If we can figure out how to do that in the future, that would be great.